process-engine/bpmn-studio
# ✨ Implement cyclic timer events

## Changes

1. Re- enable use of Cyclic TimerStartEvent definitions
2. Fix an issue that caused timer event definitions of any kind to disappear from TimerStartEvents
3. Add `enabled` extension property to TimerStartEvents that use cyclic timers
    - This will allow a user to disable a cyclic timer, without having to delete the entire ProcessModel from the remote solution

## Issues

Part of https://github.com/process-engine/process_engine_runtime/issues/118

PR: #1537

## How to test the changes

- Create a ProcessModel with cyclic TimerStartEvents
- Note that the property panel displays them as expected again
- Also note an `enabled` checkbox to be present, when switching to `cyclic` TimerStartEvents
- Deploy the ProcessModel with `enabled` set to `true`
    - The ProcessEngine will automatically start the ProcessModel, when the corresponding cronjob expires
- Deploy the ProcessModel with `enabled` set to `false`
    - The ProessEngine will no longer execute the ProcessModel automatically
- Undeploy the ProcessModel
    - The ProcessEngine will remove all cronjobs created for that ProcessModel
